### ISSUE TITLE
Issue when sending C2D message

### ISSUE THREAD

**Issue by FabianKLS** [Azure/iot-edge#289]
Hi, I'm successfully sent sensor data from the device to IoThub. When I try to send a message back to the device I'm receiving the following errors:

`Error: Time:Tue May 30 14:56:14 2017 File:/home/pi/iot-edge/deps/iot-sdk-c/iothub_client/src/uamqp_messaging.c Func:readPropertiesFromuAMQPMessage Line:242 Failed to get value of uAMQP message 'message-id' property (981).
Error: Time:Tue May 30 14:56:14 2017 File:/home/pi/iot-edge/deps/iot-sdk-c/iothub_client/src/uamqp_messaging.c Func:IoTHubMessage_CreateFromUamqpMessage Line:434 Failed reading properties of the uamqp message.
Error: Time:Tue May 30 14:56:14 2017 File:/home/pi/iot-edge/deps/iot-sdk-c/iothub_client/src/iothubtransport_amqp_messenger.c Func:on_message_received_internal_callback Line:624 on_message_received_internal_callback failed (IoTHubMessage_CreateFromUamqpMessage; error = 436).
`

I'm using iothub-explorer to send the device. 
`iothub-explorer send microbitgw01 {"type": "write_once", "characteristic_uuid": "E95D93EE-251D-470A-A062-FA1922DFA9A8", "data": "aGFsbG8="}`

I have the same instruction in the gateway config and this is successfully executed at initialization.

`{"type": "write_at_init", "characteristic_uuid": "E95D93EE-251D-470A-A062-FA1922DFA9A8", "data": "aGFsbG8="}`

The output shows that the message is sent successfully. 

`Message sent with id: e75a7839-64f3-48f1-8058-236f16014cf2`

I don't know how to debug this issue to find out what going wrong. 
Any suggestions?

**Comment by darobs** [comment]
Hello @FabianKLS 

I tried this same iothub-explorer command with the AMQP sample code in the Azure IoTHub Devices SDK and received a similar error. I then tried the same message on both the Azure portal and the [DeviceExplorer](https://github.com/Azure/azure-iot-sdk-csharp/tree/master/tools/DeviceExplorer), and I received the message as you might expect.   The iothub-explorer command is doing something different than the other tools we have to manually send data.  

I've contacted the people who support the iothub-explorer to see if they have suggestions for the command line input to make it work.
